### DISCUSSION TITLE
GitHub-native Dependabot can't merge to protect branch

### DISCUSSION THREAD

**Discussion by ZebraFlesh** [community/community#24017] [New to GitHub]
<p>I’m trying to migrate from dependabot-preview to the GitHub-native Dependabot. One issue I’m running into is branch protections: once a PR is ready to merge, I’ve tried issuing a <code>@dependabot merge</code> command. Unfortunately this runs into branch protection rules. With dependabot-preview, you could add the dependabot-preview app as an app with push access to the branch. However, I’m not seeing an equivalent app for GitHub-native Dependabot.</p>
<p>How can I get GitHub-native Dependabot to successfully execute the merge command against a protected branch?</p>

**Comment by nethgato** [comment]
<p>Hi there <a class="mention" href="/ZebraFlesh">@ZebraFlesh</a> – I love your name and av =D</p>
<p>So you should just be able to add <code>Dependabot Preview</code> to the list of users who are allowed to push (from the protected branch settings page). Note that I shamelessly cmd+c/cmd+v’d this:</p>
<aside class="onebox githubissue">
  <header class="source">
      <a href="https://github.com/dependabot/feedback/issues/677#issuecomment-533046961" target="_blank" rel="noopener">github.com/dependabot/feedback</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/dependabot/feedback/issues/677#issuecomment-533046961" target="_blank" rel="noopener">Merging after approval and protected branches</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2019-09-19" data-time="02:03:30" data-timezone="UTC">02:03AM - 19 Sep 19 UTC</span>
      </div>

        <div class="date">
          closed <span class="discourse-local-date" data-format="ll" data-date="2019-09-20" data-time="07:39:15" data-timezone="UTC">07:39AM - 20 Sep 19 UTC</span>
        </div>

      <div class="user">
        <a href="https://github.com/rtpg" target="_blank" rel="noopener">
          <img alt="rtpg" src="https://user-images.githubusercontent.com/62281706/181080645-45fa7858-9234-4757-9bbc-90c3638ecd82.jpeg" class="onebox-avatar-inline" width="20" height="20">
          rtpg
        </a>
      </div>
    </div>
  </div>
</div>

<div class="github-row">
  <p class="github-content">We have the following setup:
a non-master default branch
requiring approval before merging into this default branch
master as a protected branch
We don't have...</p>
</div>

<div class="labels">
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Can you let us know if you’re running into any particular issue in adding the bot to your protected branch’s settings?</p>

**Comment by ZebraFlesh** [comment]
<p>The <code>dependabot-preview</code> user is already allowed to push to the protected branch. However, this PR is from the <code>dependabot</code> user (AKA GitHub-native Dependabot). Searching for <code>dependabot</code> in the list of users to add to the branch protection yields a single result of <code>dependabot-preview</code>.</p>

**Comment by nethgato** [comment]
<p>Hey <a class="mention" href="/ZebraFlesh">@ZebraFlesh</a> I’m so sorry. I now realize I’m totally looking at outdated information and the post I linked to is not at all relevant anymore. Also, you straight up said you had the <code>preview</code> added as a privileged user to your branch. Yikes; not my best…</p>
<p>I think there is more information that will be helpful, though.</p>
<p>Is this a public repo? If so, got a link? If not, no worries!</p>
<p>Though the main difference is that the preview had a user that was required to have privileges. With the released Dependabot, it’s managed in the platform itself and there’s no user to add to your repositories.</p>
<p>Dependabot features are located in <code>/{owner}/{repo}/settings/security_analysis</code>.</p>
<p>I really hope this is more helpful than my first response! :crossed_fingers:</p>

**Comment by ZebraFlesh** [comment]
<p>This is a private repo. There doesn’t seem to be a setting in <code>/settings/security_analysis</code> that would allow me to grant access to protected branches.</p>

**Comment by nethgato** [comment, marked as answer]
<p>Hey <a class="mention" href="/ZebraFlesh">@ZebraFlesh</a> – after some more diggin’, it looks like auto-merges aren’t quite possible with Dependabot, as-is.</p>
<p><a href="https://github.com/dependabot/dependabot-core/issues/2268">This issue</a> discusses your options (use Actions) in more detail.</p>
<p><a href="https://github.com/dependabot/dependabot-core/issues/2268#issuecomment-654608325">This comment</a> outlines the steps to create an Actions workflow which leverages Dependabot and mentions the configuration required for private repos.</p>

**Comment by ZebraFlesh** [comment]
<p>Thanks, but that still doesn’t answer my initial question: how to get a v2 <code>@dependabot merge</code> command to succeed against a protected branch. Whether or not I’m using GitHub Actions is orthogonal to this problem: Dependabot cannot execute the command (via a PR comment) and merge to the protected branch. The comment which you highlighted is how to automate making the PR comment, but it would still fail to merge for me because of branch protections.</p>

**Comment by AlCalzone** [comment]
<p>I’ll have to agree with <a class="mention" href="/ZebraFlesh">@ZebraFlesh</a>. Using an action to tell dependabot to merge is <strong>not</strong> a solution to the issue here.<br>
In fact, I’m already doing this for all of my repos, but this fails for those which have a protected master branch. In this case, dependabot fails with the following error:</p>
<pre><code class="lang-auto">You're not authorized to push to this branch. Visit https://docs.github.com/articles/about-protected-branches/ for more information.
</code></pre>
<p>It is also <strong>not</strong> a solution to automate the merge ourselves, because dependabot takes care of auto-updating the PRs if there is a conflict with the target branch.</p>
